File: src/main/java/topwords/WordQueue.java
package topwords;

import java.util.LinkedList;
import java.util.Queue;

public class WordQueue {
    Queue<String> wordQueue;
    public WordQueue() {
        wordQueue = new LinkedList<>();
    }

    public void addWord(String w) { wordQueue.add(w);}

    public String removeWord() {
        return wordQueue.remove();
    }

    public int getQueueSize() { return wordQueue.size();}
}

File: src/main/Scala/com/topwords/WordQueue.scala
package com.topwords

class WordQueue {
  import scala.collection.mutable.Queue
  var q: Queue[String] = Queue.empty[String]

  def add(word: String, cloud: WordCloud): Unit = {
    if (word.length >= cloud.minLength) {
      q.enqueue(word)
      cloud.insertOrIncrement(word)
    }
  }

  def remove = {
    q.dequeue
  }

  def getQueueSize: Int = {
    q.length
  }
}